### DISCUSSION TITLE
Conflicting Star Exports Warnings with Next.js 13.5 and MUI

### DISCUSSION THREAD

**Discussion by lucasvidela94** [vercel/next.js#55740] [App Router]
Since the recent update to Next.js, specifically in version 13.5, I've been encountering new warnings in the terminal related to dependencies. These warnings seem to be related to the handling of barrels (modules that re-export multiple modules) and might be caused by changes in how Next.js deals with them. While these warnings don't seem to affect the functionality of my application, they are filling up the terminal and preventing to read the logs.

**Specific Errors:**

- Error 1:
- Import trace for the requested module:
  ```
  __barrel_optimize__?names=Stack,Typography,useMediaQuery,useTheme!=!./node_modules/.pnpm/@mui+material@5.14.10_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.20_react-dom@18.2.0_react@18.2.0/node_modules/@mui/material/index.js
  ./src/app/components/PageFooter.tsx
  ./src/app/login/page.tsx
  ```

- Error 2:

- Import trace for the requested module:
  ```
  __barrel_optimize__?names=Stack,Typography,useMediaQuery,useTheme!=!./node_modules/.pnpm/@mui+material@5.14.10_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.20_react-dom@18.2.0_react@18.2.0/node_modules/@mui/material/index.js
  ./src/app/components/PageFooter.tsx
  ./src/app/login/page.tsx
  ```

- [Repeat for other errors...]

These warnings are consistently appearing in the terminal, making it difficult to review other logs and debug effectively. While the application still functions correctly, did anyone have the same issue? am i missing something? 😮‍💨

**Comment by vilbert** [comment]
I recently got the same error and some sources said it's because of some circular imports happening while bundling using webpack.

Anyway, my issue came from custom styled MUI components, and here's how I fixed it:

Changed  `import { styled } from "@mui/material"`
to `import { styled } from "@mui/material/styles"`

Hope it helps.

**Comment by codemile** [comment]
@vilbert is correct. I had the same warnings and updated the imports to fix them. You can add an eslint import rule to catch these in the future.

**Comment by hotspoon** [comment]
based on @ticovix
I downgraded the NEXT version to 13.4.19 and the warnings stopped.

this is working for me, downgrade nextjs version from 13.5 to 13.4.19 remove the warning in terminal

**Comment by niteshsharma0304** [comment]
@codemile , If you have the es-lint rule for this, could you please share it?

**Reply by hobroker** [reply]
I wrote these and seems to find my problems 
```js
module.exports = {
  create(context) {
    return {
      ImportDeclaration(node) {
        if (node.source.value !== "@mui/material") return;

        node?.specifiers?.forEach((item) => {
          if (["styled", "useTheme"].includes(item.local.name)) {
            context.report(
              node,
              item.loc,
              `Import "${item.local.name}" from "@mui/material/styles" instead.`,
            );
          }
        });
      },
    };
  },
};
```

**Reply by SonNg2k** [reply]
This ESLint rule https://typescript-eslint.io/rules/no-restricted-imports/ will help u enforce this. For example:

```javascript
'no-restricted-imports': 'off',
'@typescript-eslint/no-restricted-imports': [
  'error',
  {
    paths: [
      {
        name: '@mui/material',
        message:
          "Imports from '@mui/material' are not allowed, use '@mui/material/*' instead.",
      },
      {
        name: '@mui/icons-material',
        message:
          "Imports from '@mui/icons-material' are not allowed, use '@mui/icons-material/*' instead.",
      },
    ],
  },
],
```

**Comment by ivandax** [comment]
In my case it got fixed by changing:

`import { ThemeProvider } from "@mui/material";`

to 

import ThemeProvider from "@mui/material/styles/ThemeProvider";

**Reply by asaad2121** [reply]
isnt that a syntax error?
I got this `SyntaxError: Cannot use import statement outside a module`

**Comment by sai9494** [comment]
I have one doubt in mui
<img width="570" alt="Screenshot (9)" src="https://github.com/vercel/next.js/assets/80608546/8727ad22-0289-4549-ab73-687934f7048f">
 once use variant= standard for textfiled mui is getting any flickering issue
